packages/rum/src/domain/record/serialize.ts

```diff
@@ -270,17 +374,51 @@ function shouldIgnoreElement(element: Element) {
   return false
 }
 
-function serializeTextNode(text: Text, options: SerializeOptions): TextNode | undefined {
+/**
+ * Text Nodes are dependant on Element nodes
+ * Privacy levels are set on elements so we check the parentElement of a text node
+ * for privacy level.
+ */
+function serializeTextNode(textNode: Text, options: SerializeOptions): TextNode | undefined {
   // The parent node may not be a html element which has a tagName attribute.
   // So just let it be undefined which is ok in this use case.
-  const parentTagName = text.parentNode && (text.parentNode as HTMLElement).tagName
-  let textContent = text.textContent || ''
-  const isStyle = parentTagName === 'STYLE' ? true : undefined
-  if (isStyle) {
-    textContent = makeStylesheetUrlsAbsolute(textContent, location.href)
-  } else if (options.ignoreWhiteSpace && !textContent.trim()) {
+  const parentTagName = textNode.parentElement?.tagName
+  let textContent = textNode.textContent || ''
+
+  if (options.ignoreWhiteSpace && !textContent.trim()) {
     return
   }
+
+  const nodePrivacyLevel = getNodePrivacyLevel(textNode.parentNode as Node)
```

Comment:
```suggestion
  const nodePrivacyLevel = options.parentNodePrivacyLevel
```

Reply:
Worried about privacy level changes from parent->child. Would prefer to address this with the privacy level caching spec.

Reply:
ah, I misunderstood. Yes, we should be using the `parentNodePrivacyLevel` since we descoped caching.